main.go

```diff
@@ -1,117 +1,144 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 package main
 
 import (
-	"context"
-
-	"github.com/giantswarm/microerror"
-	"github.com/giantswarm/microkit/command"
-	microserver "github.com/giantswarm/microkit/server"
-	"github.com/giantswarm/micrologger"
-	"github.com/spf13/viper"
-
-	"github.com/giantswarm/cluster-api-provider-kvm/flag"
-	"github.com/giantswarm/cluster-api-provider-kvm/pkg/project"
-	"github.com/giantswarm/cluster-api-provider-kvm/server"
-	"github.com/giantswarm/cluster-api-provider-kvm/service"
+	"flag"
+	"os"
+
+	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	// to ensure that exec-entrypoint and run can make use of them.
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/klog/klogr"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/cluster-api/feature"
+	"sigs.k8s.io/cluster-api/util/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+
+	"github.com/giantswarm/cluster-api-provider-kvm/controllers"
+	//+kubebuilder:scaffold:imports
 )
 
 var (
-	f *flag.Flag = flag.New()
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
 )
 
-func main() {
-	err := mainE(context.Background())
-	if err != nil {
-		panic(microerror.JSON(err))
-	}
+func init() {
+	klog.InitFlags(nil)
+
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+
+	//+kubebuilder:scaffold:scheme
 }
 
-func mainE(ctx context.Context) error {
-	var err error
+var (
+	metricsAddr          string
+	enableLeaderElection bool
+	probeAddr            string
+)
 
-	var logger micrologger.Logger
-	{
-		c := micrologger.Config{}
+func main() {
 
-		logger, err = micrologger.New(c)
-		if err != nil {
-			return microerror.Mask(err)
-		}
+	initFlags(pflag.CommandLine)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	pflag.Parse()
+
+	ctrl.SetLogger(klogr.New())
+	ctx := ctrl.SetupSignalHandler()
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "a0acdc7d.infrastructure.cluster.x-k8s.io",
```

Comment:
This is just a random prefix?

Reply:
This got autogenerated by kubebuilder. Changed the value to `controller-leader-election-capk` (following the pattern from other upstream operators).